### PR TITLE
fix: Make tutorial panel downloadUrl optional

### PR DESCRIPTION
**Pull request by abdhalees**
### Description

Make download link in tutorial panel optional by not rendering the link when `downloadUrl` is not passed

Related links, issue #872 

### How has this been tested?

added a unit test

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/tutorial-panel/components/tutorial-list/index.tsx [hunk 1]
```diff
@@ -65,20 +65,21 @@ export default function TutorialList({
...
+              className={styles['download-link']}
+              aria-label={i18nStrings.labelTutorialListDownloadLink}
+            >
+              <InternalIcon name="download" />
+              <InternalBox padding={{ left: 'xs' }} color="inherit" fontWeight="bold" display="inline">
+                {i18nStrings.tutorialListDownloadLinkText}
+              </InternalBox>
+            </a>
+          )}
           {/*
```
Comment (just-boris) [inline src/tutorial-panel/components/tutorial-list/index.tsx:83]: How about cleaning up commented code a bit? it does not seem like we will use it any time soon

**Review by just-boris — APPROVED** [review]

File: src/tutorial-panel/components/tutorial-list/index.tsx (hunk 1 above)
Comment (abdhalees) [reply src/tutorial-panel/components/tutorial-list/index.tsx:83]: https://github.com/cloudscape-design/components/pull/883

Comment (abdhalees) [reply src/tutorial-panel/components/tutorial-list/index.tsx:83]: removed in a separate PR as the code is more than just these lines, and to allow us to revert when we want to add that feature

Comment (just-boris) [reply src/tutorial-panel/components/tutorial-list/index.tsx:83]: Great, even more code than I expected

**Pull request merged**